### DISCUSSION TITLE
v4.1.0: Add QPALM solver

### DISCUSSION THREAD

**Discussion by stephane-caron** [qpsolvers/qpsolvers#253] [Announcements]
This release adds [QPALM](https://kul-optec.github.io/QPALM/Doxygen/), a proximal augmented Lagrangian based solver for (possibly nonconvex) quadratic programs. Check it out!

### Added

- New solver: [QPALM](https://kul-optec.github.io/QPALM/Doxygen/)
- Unit test for internal linear-box inequality combination

### Changed

- Internal: refactor linear-box inequality combination function
- Renamed main branch of the repository from `master` to `main`

### Fixed

- Fix combination of box inequalities with empty linear inequalities
- Gurobi: Account for a slight regression in QPSUT01 performance

<hr /><em>This discussion was created from the release <a href='https://github.com/qpsolvers/qpsolvers/releases/tag/v4.1.0'>v4.1.0</a>.</em>

**Comment by PierreQuinton** [comment]
Hi, this seems nice, I am eager to try it. A quick question though : In the formulation of the problem of QPALM, it seems that the domain of optimization is limited to be $b_{\min} \leq Ax \leq b_{\max}$. Can we use this solver with $b_{\max} = \infty$ if it is actually positive definite ?

**Reply by stephane-caron** [reply]
In the qpsolvers API box bounds can contain infinite values :ok_hand: That is, in:

$$
\begin{split}
\begin{array}{ll}
    & G x \leq h \\
    & A x = b \\
    & lb \leq x \leq ub
\end{array}
\end{split}
$$

We can set $lb_i = -\infty$ or $ub_i = +\infty$ to disable some box-constraint coordinates. The QP is then converted in the backend solver's format. (In the case of QPALM the conversion is straightforward as $b_{min,i} = -\infty$ or $b_{max,i} = +\infty$ are allowed.)

If you have further questions on QPALM in particular don't hesitate to head over to the [QPALM discussions](https://github.com/kul-optec/QPALM/discussions).
